pkg/policies/engine/rego/boilerplate.go

```diff
@@ -0,0 +1,163 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rego
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/open-policy-agent/opa/v1/ast"
+)
+
+const (
+	ruleResult     = "result"
+	ruleSkipped    = "skipped"
+	ruleSkipReason = "skip_reason"
+	ruleValidInput = "valid_input"
+	ruleViolations = "violations"
+	ruleIgnore     = "ignore"
+)
+
+//go:embed boilerplate.rego.tmpl
+var boilerplateTemplate string
+
+type boilerplateData struct {
+	NeedsResult     bool
+	NeedsSkipReason bool
+	NeedsSkipped    bool
+	NeedsValidInput bool
+	NeedsViolations bool
+	NeedsIgnore     bool
+}
+
+// InjectBoilerplate automatically injects common policy boilerplate if it doesn't exist.
+// This allows users to write simplified policies with only the violations rules.
+// Requirements: Policy must have package declaration and import rego.v1
+// The function:
+// - Parses the policy using OPA's AST
+// - Detects which boilerplate rules are missing
+// - Injects only the missing rules after package and imports
+func InjectBoilerplate(policySource []byte, policyName string) ([]byte, error) {
+	if len(policySource) == 0 {
+		return nil, fmt.Errorf("empty policy source")
+	}
+
+	originalPolicy := string(policySource)
+
+	// Parse the policy
+	module, err := ast.ParseModule(policyName, originalPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse policy (must have 'package' and 'import rego.v1'): %w", err)
+	}
+
+	// Detect which rules already exist using AST
+	existingRules := detectExistingRules(module)
+
+	// If all required boilerplate rules exist, no injection needed
+	if existingRules[ruleResult] && existingRules[ruleSkipReason] &&
+		existingRules[ruleSkipped] && existingRules[ruleValidInput] &&
+		existingRules[ruleViolations] && existingRules[ruleIgnore] {
+		return policySource, nil
+	}
+
+	// Build the boilerplate injection (rules only, no package/import)
+	injection, err := buildBoilerplate(existingRules)
+	if err != nil {
+		return nil, err
+	}
+
+	// If nothing needs to be injected, return original
+	if injection == "" {
+		return policySource, nil
+	}
+
+	// Inject after package and imports
+	injected, err := injectAfterImports(module, originalPolicy, injection)
+	if err != nil {
+		return nil, fmt.Errorf("failed to inject boilerplate: %w", err)
+	}
+
+	return []byte(injected), nil
+}
+
+// detectExistingRules scans the AST to find which rules are already defined
+func detectExistingRules(module *ast.Module) map[string]bool {
+	existing := make(map[string]bool)
+
+	for _, rule := range module.Rules {
+		ruleName := string(rule.Head.Name)
+		existing[ruleName] = true
+	}
+
+	return existing
+}
+
+// buildBoilerplate constructs the boilerplate template based on what's missing
+func buildBoilerplate(existingRules map[string]bool) (string, error) {
+	data := boilerplateData{
```

Comment:
Be careful, we still need the `default` rules. So maybe we should also look if there is already a "default" rule for each
```
default skipped := true
```